Update LLVM to 19.1.3

dianqk opened the pull request:
Closes #131031.

r? nikic

dianqk committed:
Update LLVM to 19.1.3

jieyouxu commented:
This doesn't have a regression test because it's a PITA (or next to impossible) to test for #131031 right?

DianQK commented:
> This doesn't have a regression test because it's a PITA (or next to impossible) to test for #131031 right?

Looks like this is worth adding a test case.

dianqk committed:
Add a regression test for #131031

The failure output is:
```
SplitVectorOperand Op #1: t51: i32 = llvm.wasm.alltrue TargetConstant:i32<12408>, t50

rustc-LLVM ERROR: Do not know how to split this operator's operand!
```

jieyouxu commented on tests/ui/target-feature/wasm-disable-simd.rs:
```diff
@@ -0,0 +1,12 @@
+//@ only-wasm32-wasip1
+//@ compile-flags: -Ctarget-feature=-simd128 --crate-type=lib
```
Question: does this need `//@ needs-llvm-component: webassembly`?

DianQK replied:
I don't think it's needed. This is my copy from https://github.com/rust-lang/rust/blob/80b74d397fffc9deaafd2b44552ba3a9107b4d5b/tests/ui/target-feature/wasm-relaxed-simd.rs .

nikic commented:
@bors r+

bors commented:
:pushpin: Commit ebeaaadceae400e371dc592b0ee89d1913d40a03 has been approved by `nikic`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=nikic ebeaaadceae400e371dc592b0ee89d1913d40a03 -->
<!-- homu: {"type":"Approved","sha":"ebeaaadceae400e371dc592b0ee89d1913d40a03","approver":"nikic","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

Zalathar commented:
Not rolling up LLVM updates seems like a good idea in general.

@bors rollup=never

bors commented:
:hourglass: Testing commit ebeaaadceae400e371dc592b0ee89d1913d40a03 with merge b2420d323786bfad4eeb42477b2fedb00ab770be...
<!-- homu: {"type":"BuildStarted","head_sha":"ebeaaadceae400e371dc592b0ee89d1913d40a03","merge_sha":"b2420d323786bfad4eeb42477b2fedb00ab770be"} -->

rust-log-analyzer commented:
The job **`dist-x86_64-illumos`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/11628769392/job/32384520961) [(plain)](https://github.com/rust-lang-ci/rust/commit/b2420d323786bfad4eeb42477b2fedb00ab770be/checks/32384520961/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11628769392/job/32394215927)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/11628769392/job/32394215927","builder_name":"checks-actions"} -->

nikic commented:
@bors retry LLVM rebuild timeout

bors commented:
:hourglass: Testing commit ebeaaadceae400e371dc592b0ee89d1913d40a03 with merge b5f4883a06f1d861b108ac892f07f4fa37eb1bed...
<!-- homu: {"type":"BuildStarted","head_sha":"ebeaaadceae400e371dc592b0ee89d1913d40a03","merge_sha":"b5f4883a06f1d861b108ac892f07f4fa37eb1bed"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/11640052562/job/32419517890)
Approved by: nikic
Pushing b5f4883a06f1d861b108ac892f07f4fa37eb1bed to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"nikic","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/11640052562/job/32419517890"},"merge_sha":"b5f4883a06f1d861b108ac892f07f4fa37eb1bed"} -->

rust-timer commented:
Finished benchmarking commit (b5f4883a06f1d861b108ac892f07f4fa37eb1bed): [comparison URL](https://perf.rust-lang.org/compare.html?start=ef972a346668ed4234d1a43ed4ad7ca4e9c58d51&end=b5f4883a06f1d861b108ac892f07f4fa37eb1bed&stat=instructions:u).

### Overall result: no relevant changes - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=ef972a346668ed4234d1a43ed4ad7ca4e9c58d51&end=b5f4883a06f1d861b108ac892f07f4fa37eb1bed&stat=instructions:u)
This benchmark run did not return any relevant results for this metric.

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=ef972a346668ed4234d1a43ed4ad7ca4e9c58d51&end=b5f4883a06f1d861b108ac892f07f4fa37eb1bed&stat=max-rss)
<details>
<summary>Results (secondary 4.1%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 4.1% | [4.1%, 4.1%] | 1     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=ef972a346668ed4234d1a43ed4ad7ca4e9c58d51&end=b5f4883a06f1d861b108ac892f07f4fa37eb1bed&stat=cycles:u)
<details>
<summary>Results (secondary -2.2%)</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -2.2% | [-2.2%, -2.2%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |
</details>

### [Binary size](https://perf.rust-lang.org/compare.html?start=ef972a346668ed4234d1a43ed4ad7ca4e9c58d51&end=b5f4883a06f1d861b108ac892f07f4fa37eb1bed&stat=size:linked_artifact)
This benchmark run did not return any relevant results for this metric.

**Bootstrap**: 779.182s -> 779.165s (-0.00%)
**Artifact size**: 335.34 MiB -> 335.32 MiB (-0.00%)
